openedx/core/djangoapps/user_api/accounts/permissions.py

```diff
@@ -16,6 +16,16 @@ def has_permission(self, request, view):
         return request.user.has_perm('student.can_deactivate_users')
 
 
+class CanCancelUserRetirement(permissions.BasePermission):
+    """
+    Grants access to cancel retirement if the requesting user is a superuser,
+    or has the explicit permission to cancel retirement of a User account.
+    """
+
+    def has_permission(self, request, view):
+        return request.user.is_superuser or request.user.has_perm('user_api.change_userretirementstatus')
```

Comment:
Instead of OR, shouldn't it be AND? (is_staff OR is_superuser) AND has_perm? 
Is there a doc outlining what set of users can cancel retirement?

Reply:
Basically, anyone having access to change UserRetirementStatus can cancel retirement: https://2u-internal.atlassian.net/wiki/spaces/SPRT/pages/14093581/Account+Un-Deletion+Process#Step-4:-No-results

Reply:
Ok, then it should be superuser or (staff who has perm to change and delete user retirement).